File: F/2351-first-letter-to-appear-twice/main.go
package main

func repeatedCharacter(s string) byte {
	buf := make([]int, 26)
	for _, b := range s {
		offset := byte(b) - 'a'
		if buf[offset] == 1 {
			return byte(b)
		}
		buf[offset]++
	}
	return '0'
}

File: F/2351-first-letter-to-appear-twice/main.rs
struct Solution {}

impl Solution {
  pub fn repeated_character(s: String) -> char {
    let mut buf: [i32; 26] = [0; 26];
    for c in s.chars() {
      let offset = c - 'a' as u8;
      buf[offset] += 1;
      if buf[offset] == 2 {
        return c;
      }
    }
    'a'
  }
}
